src/frontend/src/eth/services/send.services.ts

```diff
@@ -508,14 +511,45 @@ const approve = async ({
 			erc20HelperContractAddress
 		})
 	) {
-		return { transactionApproved: false };
+		return { transactionApproved: false, nonce };
 	}
 
+	const preApprovedAmount = await erc20ContractAllowance({
+		token,
+		owner: from,
+		spender: erc20HelperContractAddress,
+		networkId
+	});
+
+	// If there is already an approved allowance that is enough for the required amount, we don't need to approve again.
+	if (preApprovedAmount.gte(amount)) {
+		progress(ProgressStepsSend.APPROVE);
+		return { transactionApproved: false, nonce };
+	}
+
+	// If the existing pre-approved amount is not enough but non-null, we need to reset the allowance first and then approve the new amount.
+	if (preApprovedAmount.gt(ZERO)) {
+		await prepareAndSignApproval({
```

Comment:
1. For readability maybe a local function?

```
const resetExistingApprovalToZero = async () => await prepareAndSignApproval({...

await resetExistingApprovalToZero();
```

2. `prepareAndSignApproval` update the progression in the UI, is it correct to do so at this stage? Isn't the UI going to go back an forth between steps SIGN_APPROVE -> APPROVE -> SIGN_APPROVE -> APPROVE?

Reply:
1. Yes, I was wondering that myself, it is cleaner

2. Not really, because we are not passing `progress` to the `reset` function. So, in the UI, the difference is that the initialization would take more time (I tested it), before going to `SIGN_APPROVE`.

I was thinking of making it with more steps, if we need to reset it. But it is not that easy: conditional steps that change while the progress is already "in progress".

What do you suggest? Ideally, I could just change the label of the INITIALIZATION step: it would dynamically change from "Initializing transaction" to "Initializing transaction (reset allowance)", or something similar, boh

Reply:
I'm not sure UX wise that we should go too deep in the details. Since we are not passing `progress` I guess my concern is already addressed.

Reply:
cool!